lib/chef/knife/bootstrap.rb

```diff
@@ -375,9 +375,7 @@ def run
             exit 1
           end
 
-          client_builder.run
-
-          chef_vault_handler.run(node_name: config[:chef_node_name])
+          chef_vault_handler.run(client_builder.run[:client])
```

Comment:
Returning an arbitrary hash from run is smelly. I think the pattern we use is that run tells the object to go do its thing and all we ever return from run is an indicator of success/failure.

Here's a similar example: https://github.com/chef/chef/blob/master/lib/chef/client.rb#L713-L716

Imagine a more extreme case where we returned 15 more properties from a run method that the object generated in a hash. Now you've got the class representing the object and this other hash that you're looking through. It'd be pretty silly to have a `person` object as well as a separate `things_we_know_about_person` hash.

It seems like there's a perfectly good ClientBuilder object there to use and we should get its properties from it rather then storing them in local variables.